### PR TITLE
nitdoc: migrate the DocPage rendering to composite rendering

### PR DESCRIPTION
**Pull request by Morriar**
Also delete old nitdoc components.

### CODE REVIEW

File: src/doc/doc_phases/doc_html.nit [hunk 1]
```diff
@@ -532,267 +464,142 @@ redef class DocComposite
...
-	# Start the rendering from root.
-	#
-	# FIXME needed to maintain TplSection compatibility.
-	fun start_rendering(v: RenderHTMLPhase, doc: DocModel, page: MEntityPage): TplSection do
-		var section = new TplSection("top")
-		var mentity = page.mentity
-		section.title = mentity.html_name
-		section.subtitle = mentity.html_declaration
-		# FIXME ugly hack to avoid diff
+# FIXME hideous hacks to avoid diff
```
Comment (R4PaSs) [inline src/doc/doc_phases/doc_html.nit]: So, we left the ugly hack to find a collection of hideous hacks ?
Sounds like proper improvement from my point of view :)

Comment (Morriar) [inline src/doc/doc_phases/doc_html.nit]: When there is ugly hack it's to follow the current page organization and limit the diff.

It makes it easier to compare two successives versions of the doc. The removing of these hacks will be done in further PRs.

File: src/doc/doc_phases/doc_html.nit [hunk 2]
```diff
@@ -435,47 +351,68 @@ redef class MClassPage
...
 
-	private fun tpl_sidebar_item(mprop: MProperty): TplListItem do
+	private fun tpl_sidebar_item(mprop: MProperty): ListItem do
 		var classes = mprop.intro.css_classes
 		if not mprop_is_local(mprop) then
 			classes.add "inherit"
 			var cls_url = mprop.intro.mclassdef.mclass.nitdoc_url
-			var def_url = "{cls_url}#{mprop.nitdoc_id}"
-			var lnk = new TplLink(def_url, mprop.html_name)
+			var def_url = "{cls_url}#article:{mprop.nitdoc_id}.definition"
```
Comment (privat) [inline src/doc/doc_phases/doc_html.nit]: je me demande si un service dans mentity pourrait pas factoriser les "article:x.definition".

~~~
var def_url = "{cls_url}#{mprop.definition_id}"
~~~

Comment (Morriar) [inline src/doc/doc_phases/doc_html.nit]: Pas dans MEntity car une même MEntity peut être utilisée dans une ConcernSection, un DefinitionArticle ou un InheritanceGraphArticle.

**Comment by privat** [comment]
+1

File: src/doc/html_templates/html_templates.nit [hunk 3]
```diff
@@ -147,6 +125,16 @@ redef class DocPage
 		addn "</div>"
 		render_footer
 	end
+
+	# Render table of content fir this page.
```
Comment (R4PaSs) [inline src/doc/html_templates/html_templates.nit]: *first

Comment (R4PaSs) [inline src/doc/html_templates/html_templates.nit]: *for

**Comment by R4PaSs** [comment]
:shipit:

**Comment by Morriar** [comment]
Reroll for the typo as raised by @R4PaSs!

**Pull request merged**